### PR TITLE
Updates contributing guide regarding signals, removes SIGNAL_HANDLER_DOES_SLEEP

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Updates the contribution to include a section on the usage of signals by requiring the use of `SIGNAL_HANDLER`, preferring a comment to mark subscribed signals, and explaining how to invoke async instead of just skipping the handler. Also cleans up some things the markdown linter was complaining about.

In addition, this removes `SIGNAL_HANDLER_DOES_SLEEP` from the codebase. It was a legacy addition that was now only used in one place, and rather than have to explain "this exists, don't use it" in the documentation, I figured it made more sense to just nuke it.

## Why It's Good For The Game
We don't really have a lot of documentation around best-practice usages for signals. This should hopefully formalize some requirements, and make sure we're smart about their usages going forward. These should both make signal code easier to read/review, as well as hopefully helping us to catch some errors before they happen.

Requiring a comment noting signal sources is something that I started doing a while back mostly for my own bookkeeping, but it seems like something that should hopefully help to avoid the spaghetti that can occur when you're juggling multiple signals with many signal handlers. Having a proc refer back to the signal source helps to clarify its arguments, and also helps you to find which handler is responsible for which signal (and vice versa) without having to search for where it was registered.

## Testing
Verified the changes to the heart worked properly.

## Changelog
NPFC

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -154,18 +154,18 @@ Do not compare boolean values to TRUE or FALSE. For TRUE you should just check i
 // Bad
 var/thing = pick(list(TRUE, FALSE))
```
Comment (DGamerL) [inline .github/CONTRIBUTING.md:155]: Not sure if it's in scope to change this, but we don't need to put things in lists for `pick()` to work

**Review by DGamerL — APPROVED** [review]
LGTM, I don't think this needs a Bad example like the other sections of the file

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Contrabang) [reply .github/CONTRIBUTING.md:155]: Yeah, I think its an example of bad code

Comment (lewcc) [reply .github/CONTRIBUTING.md:155]: Yeah, might want to do that in a separate PR, could probably be worth a once over once this goes in

Comment (DGamerL) [reply .github/CONTRIBUTING.md:155]: I'll open a PR to change it tonight then

**Review by Henri215 — APPROVED** [review]

**Review by Contrabang — DISMISSED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -814,6 +840,7 @@ Each role inherits the lower role's responsibilities (IE: Headcoders also have c
 
 `Commit Access` members have write access to the repository and can merge your PRs. People included in this role are:
 
+
```
Comment (DGamerL) [inline .github/CONTRIBUTING.md:843]:
```suggestion
```
Yeah just do this and it should be good to go

**Review by FunnyMan3595 — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**